internal/config/configdomain/use_message_test.go

```diff
@@ -0,0 +1,149 @@
+package configdomain_test
+
+import (
+	"testing"
+
+	"github.com/git-town/git-town/v17/internal/config/configdomain"
+	"github.com/git-town/git-town/v17/internal/git/gitdomain"
+	. "github.com/git-town/git-town/v17/pkg/prelude"
+	"github.com/shoenig/test/must"
+)
+
+func TestUseMessage(t *testing.T) {
+	t.Parallel()
+
+	t.Run("EditDefaultMessage", func(t *testing.T) {
+		t.Parallel()
+
+		m := configdomain.EditDefaultMessage()
+
+		// Custom type getters.
+		_, isCustom := m.GetCustomMessage()
+		must.False(t, isCustom)
+		// Type checks.
+		must.False(t, m.IsCustomMessage())
+		must.True(t, m.IsEditDefault())
+		must.False(t, m.IsUseDefault())
+	})
+
+	t.Run("GetCustomMessageOrPanic", func(t *testing.T) {
+		// Arrange.
+		t.Parallel()
+		m := configdomain.EditDefaultMessage()
+
+		// Assert.
+		defer func() {
+			r := recover()
+			must.Eq(t, r, "UseMessage is not UseCustomMessage")
+		}()
+
+		// Act.
+		_ = m.GetCustomMessageOrPanic()
+	})
+
+	t.Run("UseCustomMessage", func(t *testing.T) {
+		t.Parallel()
+		want := gitdomain.CommitMessage("foo")
+
+		m := configdomain.UseCustomMessage(want)
+
+		// Custom type getters.
+		message, isCustom := m.GetCustomMessage()
+		must.True(t, isCustom)
+		must.Eq(t, want, message)
+		must.Eq(t, want, m.GetCustomMessageOrPanic())
+		// Type checks.
+		must.True(t, m.IsCustomMessage())
+		must.False(t, m.IsEditDefault())
+		must.False(t, m.IsUseDefault())
+	})
+
+	t.Run("UseCustomMessageOrWithSome", func(t *testing.T) {
```

Comment:
There seem to be multiple tests that verify `configdomain.UseCustomMessageOr`. How do you feel about moving them into a nested `t.Run` block?

```go
t.Run("UseCustomMessageOr", func(t *testing.T) {
  t.Run("with some", func(t *testing.T) { ... })
  t.Run("with none", func(t *testing.T) { ... })  
})
```

What I like about this setup is that it helps get an overview of the various test cases, and spot missing ones.

Reply:
Done.